corehq/apps/hqcase/api/get_list.py

```diff
@@ -128,11 +129,18 @@ def get_list(domain, couch_user, params):
 
 def _get_cursor_query(domain, params, last_date, last_id):
     query = _get_query(domain, params)
+    id_filter = queries.nested(
+        CASE_PROPERTIES_PATH,
+        filters.AND(
+            filters.term(case_search.PROPERTY_KEY, '@case_id'),
+            filters.range_filter(case_search.PROPERTY_VALUE_EXACT, gt=last_id),
+        )
+    )
```

Comment:
have we already reindexed this index? It seems a pity to have to do a nested filter for the document ID.

Reply:
Some other places that may also need to change:
* https://github.com/dimagi/commcare-hq/blob/46d76174a8784bec5845cdcf2bab17eb01f47ad9/corehq/apps/es/case_search.py#L442 (not sure if `_id` is still returned in the response from ES)
* https://github.com/dimagi/commcare-hq/blob/be3902c5cc1b9edce11dbdb4931b9a689def4b1c/corehq/apps/case_search/const.py#L44 (specifically the `sort_property` field)
*

Reply:
> have we already reindexed this index? It seems a pity to have to do a nested filter for the document ID.

@snopoke  Yeah. We have already reindexed this index. Yeah it feels bit weird to do so, we can fix this in the next reindex we do.


> (not sure if _id is still returned in the response from ES)

@snopoke Yes `_id` is still returned in the response. ES 5 just don't allow sorting and using range queries on `_id` field because of performance reasons.


> [commcare-hq/corehq/apps/case_search/const.py](https://github.com/dimagi/commcare-hq/blob/be3902c5cc1b9edce11dbdb4931b9a689def4b1c/corehq/apps/case_search/const.py#L44)
(specifically the sort_property field) 


I searched in code base where we are using the `sort_property` and found that we are only using it in `case_list_explorer` and `case_explorer`. Since `case_explorer` did not have `@case_id` so we just need to ensure that when user search on `@case_id` the query goes through.

I have added the sorting block in https://github.com/dimagi/commcare-hq/pull/33451/commits/dbbe69e874bf302557f856efd8b600d9447fb738 and tested it locally. Without the change the query was erroring out in ES 5 and UI was not getting updated and with the change the sort worked perfectly.

Reply:
Yeah, it would be really nice to establish a convention of adding `doc_id` to pillows and mappings as a matter of practice to avoid all the hacks associated with using `_id`.  I believe we manually insert `_id` into the return values for backwards compatibility here:

https://github.com/dimagi/commcare-hq/blob/8c6cddad91b2b8d241214a6ab30fa7e556d655cc/corehq/apps/es/client.py#L940-L949

I thought that was done elsewhere, but it's possible it changed or I'm misremembering.  Anyways, I added that suggestion to the list of [proposed mapping changes](https://docs.google.com/document/d/1JkiKttC1XwNiB_p8eIkS0rilvhz9gWAszEQchL6AMJo/edit)